src/VkDispatchTable.h

```diff
@@ -0,0 +1,4617 @@
+	PFN_vkCmdResetEvent fp_vkCmdResetEvent = nullptr;
+	PFN_vkCmdWaitEvents fp_vkCmdWaitEvents = nullptr;
+	PFN_vkCmdPipelineBarrier fp_vkCmdPipelineBarrier = nullptr;
+	PFN_vkCmdBeginQuery fp_vkCmdBeginQuery = nullptr;
+	PFN_vkCmdEndQuery fp_vkCmdEndQuery = nullptr;
+	PFN_vkCmdResetQueryPool fp_vkCmdResetQueryPool = nullptr;
+	PFN_vkCmdWriteTimestamp fp_vkCmdWriteTimestamp = nullptr;
+	PFN_vkCmdCopyQueryPoolResults fp_vkCmdCopyQueryPoolResults = nullptr;
+	PFN_vkCmdPushConstants fp_vkCmdPushConstants = nullptr;
+	PFN_vkCmdBeginRenderPass fp_vkCmdBeginRenderPass = nullptr;
+	PFN_vkCmdNextSubpass fp_vkCmdNextSubpass = nullptr;
+	PFN_vkCmdEndRenderPass fp_vkCmdEndRenderPass = nullptr;
+	PFN_vkCmdExecuteCommands fp_vkCmdExecuteCommands = nullptr;
+#endif
+#ifdef VK_VERSION_1_1
+	PFN_vkBindBufferMemory2 fp_vkBindBufferMemory2 = nullptr;
+	PFN_vkBindImageMemory2 fp_vkBindImageMemory2 = nullptr;
+	PFN_vkGetDeviceGroupPeerMemoryFeatures fp_vkGetDeviceGroupPeerMemoryFeatures = nullptr;
+	PFN_vkCmdSetDeviceMask fp_vkCmdSetDeviceMask = nullptr;
+	PFN_vkCmdDispatchBase fp_vkCmdDispatchBase = nullptr;
+	PFN_vkGetImageMemoryRequirements2 fp_vkGetImageMemoryRequirements2 = nullptr;
+	PFN_vkGetBufferMemoryRequirements2 fp_vkGetBufferMemoryRequirements2 = nullptr;
+	PFN_vkGetImageSparseMemoryRequirements2 fp_vkGetImageSparseMemoryRequirements2 = nullptr;
+	PFN_vkTrimCommandPool fp_vkTrimCommandPool = nullptr;
+	PFN_vkGetDeviceQueue2 fp_vkGetDeviceQueue2 = nullptr;
+	PFN_vkCreateSamplerYcbcrConversion fp_vkCreateSamplerYcbcrConversion = nullptr;
+	PFN_vkDestroySamplerYcbcrConversion fp_vkDestroySamplerYcbcrConversion = nullptr;
+	PFN_vkCreateDescriptorUpdateTemplate fp_vkCreateDescriptorUpdateTemplate = nullptr;
+	PFN_vkDestroyDescriptorUpdateTemplate fp_vkDestroyDescriptorUpdateTemplate = nullptr;
+	PFN_vkUpdateDescriptorSetWithTemplate fp_vkUpdateDescriptorSetWithTemplate = nullptr;
+	PFN_vkGetDescriptorSetLayoutSupport fp_vkGetDescriptorSetLayoutSupport = nullptr;
+#endif
+#ifdef VK_VERSION_1_2
+	PFN_vkCmdDrawIndirectCount fp_vkCmdDrawIndirectCount = nullptr;
+	PFN_vkCmdDrawIndexedIndirectCount fp_vkCmdDrawIndexedIndirectCount = nullptr;
+	PFN_vkCreateRenderPass2 fp_vkCreateRenderPass2 = nullptr;
+	PFN_vkCmdBeginRenderPass2 fp_vkCmdBeginRenderPass2 = nullptr;
+	PFN_vkCmdNextSubpass2 fp_vkCmdNextSubpass2 = nullptr;
+	PFN_vkCmdEndRenderPass2 fp_vkCmdEndRenderPass2 = nullptr;
+	PFN_vkResetQueryPool fp_vkResetQueryPool = nullptr;
+	PFN_vkGetSemaphoreCounterValue fp_vkGetSemaphoreCounterValue = nullptr;
+	PFN_vkWaitSemaphores fp_vkWaitSemaphores = nullptr;
+	PFN_vkSignalSemaphore fp_vkSignalSemaphore = nullptr;
+	PFN_vkGetBufferDeviceAddress fp_vkGetBufferDeviceAddress = nullptr;
+	PFN_vkGetBufferOpaqueCaptureAddress fp_vkGetBufferOpaqueCaptureAddress = nullptr;
+	PFN_vkGetDeviceMemoryOpaqueCaptureAddress fp_vkGetDeviceMemoryOpaqueCaptureAddress = nullptr;
+#endif
+#if defined(VK_KHR_swapchain) && defined(VK_VERSION_1_0)
+#ifndef PFN_vkCreateSwapchainKHR_DECLARE
```

Comment:
Not sure what the macro guards here accomplish. Same question as the LOAD and PROXY guards elsewhere.

Reply:
Some extension nodes had duplicate proto definitions from other extensions. `VK_KHR_swapchain` for example (line `11452`, cant link line directly because file is too large) has a require node with duplicate proto's from `VK_KHR_device_group`. The guards are to prevent multiple definitions from niche situations like this. I only see this happening twice throughout all of vk.xml, but you never know when things might change/be added.